apps/api/src/auth/auth.controller.ts

```diff
@@ -1,35 +1,64 @@
-import { Controller, Get, Logger, UseGuards } from '@nestjs/common';
 import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  InternalServerErrorException,
+  Logger,
+  Post,
+  UnauthorizedException,
+  UseGuards,
+} from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBearerAuth,
+  ApiBody,
+  ApiCreatedResponse,
   ApiMovedPermanentlyResponse,
+  ApiNoContentResponse,
+  ApiOkResponse,
   ApiOperation,
   ApiQuery,
   ApiTags,
   ApiUnauthorizedResponse,
 } from '@nestjs/swagger';
-import { AccessTokenResponse } from 'types';
+import { AccessTokenResponse, TfaNeededResponse } from 'types';
+import { UsersService } from '../users/users.service';
+import { AccessTokenResponseDto } from './access-token-response.dto';
 import { AuthService } from './auth.service';
+import { CheckTfaTokenStateDto } from './check-tfa-token-state.dto';
+import { CheckTfaTokenDto } from './check-tfa-token.dto';
 import { FtOauth2AuthGuard } from './ft-oauth2-auth.guard';
 import { FtOauth2Dto } from './ft-oauth2.dto';
-import { StateGuard } from './state.guard';
+import { JwtAuthGuard } from './jwt-auth.guard';
+import { SpeakeasyGeneratedSecretDto } from './speakeasy-generated-secret.dto';
 import { User as UserEntity } from '../users/user.entity';
 import { User } from '../common/decorators/user.decorator';
+import { StateGetGuard } from './state-get.guard';
+import { StatePostGuard } from './state-post.guard';
+import { State } from '../common/decorators/state.decorator';
+import { State as StateEntity } from './state.entity';
 
 @ApiTags('auth')
 @Controller('auth')
 export class AuthController {
   constructor(
     private readonly authService: AuthService,
+    private readonly usersService: UsersService,
     readonly logger: Logger,
   ) {}
 
   @Get('/login/oauth2/42')
-  @UseGuards(StateGuard)
+  @UseGuards(StateGetGuard)
```

Comment:
Is using two @UseGuards() working ?
Maybe use one @UseGuards() with comma separated guards